hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonFileInputFormat.java

```diff
@@ -208,6 +215,18 @@ public CarbonTable getOrCreateCarbonTable(Configuration configuration) throws IO
     return carbonFiles;
   }
 
+  private List<CarbonFile> getAllCarbonDataFiles(List fileLists) {
+    List<CarbonFile> carbonFiles = new LinkedList<CarbonFile>();
+    try {
+      for (int i = 0; i < fileLists.size(); i++) {
+        carbonFiles.add(FileFactory.getCarbonFile(fileLists.get(i).toString()));
```

Comment:
1. Need to pass hadoopConf to work with s3 ?
2. Have you checked S3 performance with this and without this for same folder read ? I guess FileFactory.getCarbonFile will slowdown S3 performance

Reply:
1. No hadoopConf can work, and old code also haven't  hadoopConf: getAllCarbonDataFiles(carbonTable.getTablePath())
2.  folder also need list file and getCarbonFIle one by one: org.apache.carbondata.core.datastore.filesystem.AbstractDFSCarbonFile#listFiles(boolean, org.apache.carbondata.core.datastore.filesystem.CarbonFileFilter)

```
  public List<CarbonFile> listFiles(boolean recursive, CarbonFileFilter fileFilter)
      throws IOException {
    List<CarbonFile> carbonFiles = new ArrayList<>();
    if (null != fileStatus && fileStatus.isDirectory()) {
      RemoteIterator<LocatedFileStatus> listStatus = fs.listFiles(fileStatus.getPath(), recursive);
      while (listStatus.hasNext()) {
        LocatedFileStatus locatedFileStatus = listStatus.next();
        CarbonFile carbonFile = FileFactory.getCarbonFile(locatedFileStatus.getPath().toString());
        if (fileFilter.accept(carbonFile)) {
          carbonFiles.add(carbonFile);
        }
      }
    }
    return carbonFiles;
  }
```